File: kedane_algo_implment/kadane_algo_implement.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_so_far = nums[0], sum = 0;
        
        for (int n : nums) {
            if (sum < 0)
                sum = 0;
            sum += n;
            max_so_far = max(max_so_far, sum);
        }
        return max_so_far;
        
    }
};

int main (){
    int i,n,el;
    cin>>n;
    vector<int> nums;
    for (i=0;i<n;i++){
        cin>>el;
        nums.push_back(el);
    }
    Solution sol;
    int res = sol.maxSubArray(nums);
    cout<<res;
    return 0;
}

File: kedane_algo_implment/kadane_algo_implement.py
class Solution(object):
    def maxSubarr(self,nums):
        print(nums)
        max_so_far=nums[0]
        sum=0
        for n in nums:
            if sum < 0:
                sum = 0
            sum += n
            max_so_far = max(max_so_far, sum)
        return max_so_far


if __name__ == "__main__":
    nums = [-2,1,-3,4,-1,2,1,-5,4]
    sol = Solution()
    res = sol.maxSubarr(nums)
    print(res)